usecases/schema/tenant.go

```diff
@@ -240,10 +240,10 @@ func (h *Handler) getTenants(class string) ([]*models.Tenant, error) {
 
 	ts := make([]*models.Tenant, info.Tenants)
 	f := func(_ *models.Class, ss *sharding.State) error {
-		if N := len(ss.Physical); N > len(ts) {
-			ts = make([]*models.Tenant, N)
-		} else if N < len(ts) {
-			ts = ts[:N]
+		if n := len(ss.Physical); n > len(ts) {
+			ts = make([]*models.Tenant, n)
+		} else if n < len(ts) {
+			ts = ts[:n]
```

Comment:
also here, can we first deduce the size and then create `ts` slice with appropriate size in one go?

Reply:
Hmm, I did not write this code @redouan-rhazouani did. But from the looks of it, I think not. Since the size of `ss.Physical` is only known within the context of `f` and `f` is executed outside of `getTenants` within `metaReader.Read`, I don't see a way `getTenants` could know the correct size of the returned object before `f` is invoked. This process looks to be the most general approach. WDYT @redouan-rhazouani?

Reply:
hmm, OK, we can leave this as it is